socorro/external/boto/connection_context.py

```diff
@@ -0,0 +1,245 @@
+        """
+        version = 'v1'
+        return '%s/%s/%s/%s' % (prefix, version, name_of_thing, id)
+
+    #--------------------------------------------------------------------------
+    def _get_bucket(self, conn, bucket_name):
+        try:
+            return self._bucket_cache
+        except AttributeError:
+            self._bucket_cache = conn.get_bucket(bucket_name)
+            return self._bucket_cache
+
+    #--------------------------------------------------------------------------
+    def _get_or_create_bucket(self, conn, bucket_name):
+        try:
+            return self._bucket_cache
+        except AttributeError:
+            try:
+                self._bucket_cache = conn.get_bucket(bucket_name)
+            except self._S3ResponseError:
+                self._bucket_cache = conn.create_bucket(bucket_name)
+            return self._bucket_cache
+
+    #--------------------------------------------------------------------------
+    def submit_to_boto_s3(self, id, name_of_thing, thing):
+        """submit something to boto.
+        """
+        if not isinstance(thing, basestring):
+            raise Exception('can only submit strings to boto')
+
+        conn = self._connect()
+        bucket = self._get_or_create_bucket(conn, self.config.bucket_name)
+
+        key = self.build_s3_dirs(self.config.prefix, name_of_thing, id)
+
+        storage_key = bucket.new_key(key)
+        storage_key.set_contents_from_string(thing)
+
+    #--------------------------------------------------------------------------
+    def fetch_from_boto_s3(self, id, name_of_thing):
+        """retrieve something from boto.
+        """
+        conn = self._connect()
+        bucket = self._get_bucket(conn, self.config.bucket_name)
+
+        key = self.build_s3_dirs(self.config.prefix, name_of_thing, id)
+
+        storage_key = bucket.get_key(key)
+        if storage_key is None:
+            raise BotoNotFound('%s not found, no value returned' % id)
+        return storage_key.get_contents_as_string()
+
+    #--------------------------------------------------------------------------
+    def _connect(self):
+        try:
+            return self.connection
+        except AttributeError:
+            kwargs = {
+                "aws_access_key_id": self.config.access_key,
+                "aws_secret_access_key": self.config.secret_access_key,
+                "is_secure": True,
+                "calling_format": self._calling_format(),
+            }
+            if self.config.host:
+                kwargs["host"] = self.config.host
+            if self.config.port:
+                kwargs["port"] = self.config.port
+            self.connection = self._connect_to_endpoint(**kwargs)
+            return self.connection
+
+    #--------------------------------------------------------------------------
+    def _convert_mapping_to_string(self, a_mapping):
+        self._stringify_dates_in_dict(a_mapping)
```

Comment:
Let's do that properly instead with passing an encoder class to `json.dumps`. That way you get it recursive and not just taking care of `datetime.datetime` objects on the "first" level. 

Also, why not isoformat()?